tests/test_classification.py

```diff
@@ -671,3 +671,81 @@ def test_bad_xval_float_classes():
 
     yield check_bad_xval_float_classes, True
     yield check_bad_xval_float_classes, False
+
+
+def check_train_and_score_function(model_type):
+    """
+    Check that the _train_and_score() function works as expected
+    """
+
+    # create train and test data
+    (train_fs,
+     test_fs) = make_classification_data(num_examples=500,
+                                         train_test_ratio=0.7,
+                                         num_features=5,
+                                         use_feature_hashing=False,
+                                         non_negative=True)
+
+    # call _train_and_score() on this data
+    estimator_name = 'LogisticRegression' if model_type == 'classifier' else 'Ridge'
+    metric = 'accuracy' if model_type == 'classifier' else 'pearson'
+    learner1 = Learner(estimator_name)
+    train_score1, test_score1 = _train_and_score(learner1, train_fs, test_fs, metric)
+
+    # this should yield identical results when training another instance
+    # of the same learner without grid search and shuffling and evaluating
+    # that instance on the train and the test set
+    learner2 = Learner(estimator_name)
+    learner2.train(train_fs, grid_search=False, shuffle=False)
+    train_score2 = learner2.evaluate(train_fs, output_metrics=[metric])[-1][metric]
+    test_score2 = learner2.evaluate(test_fs, output_metrics=[metric])[-1][metric]
+
+    eq_(train_score1, train_score2)
+    eq_(test_score1, test_score2)
+
+
+def test_train_and_score_function():
+    yield check_train_and_score_function, 'classifier'
+    yield check_train_and_score_function, 'regressor'
+
+
+@raises(ValueError)
+def test_learner_api_grid_search_no_objective():
```

Comment:
It seems a little counter-intuitive to me to have two default values ('grid_search=True' and 'grid_objective=None') conflict. Is it not best practice for a method call with all default values to run successfully? I'm not sure if this would mean add a default grid_objective or turn off grid_search by default (I think I prefer the latter). I know this isn't the point of this PR but just a thought...

Reply:
Yes, that's a good point and I agree with you. Adding a default grid objective is not an option since that's what led to this in the first place (see PR #458). However, making `grid_search` have the default value of `False` would really break a _lot_ of things for people's code using this particular API the old way (including perhaps RSMTool). And the change would be silent - as in the code would run but just _without_ grid search. I was trying to minimize that.

Reply:
Based on discussion with the other project members, I think in this case we will incur the penalty of having the default values for `grid_search` and `grid_objective` conflict. For folks who were just doing `learner.train()` without specifying anything else, the error will be very early on in the process and they will now have to explicitly think about the metric that they were using which might be a good thing. Plus the code would not just run without grid search which could seriously impact downstream performance and that too, silently. 

A suggestion was made that maybe if `grid_search` is not explicitly specified, a warning can be shown that says it is being set to `True`. That way, in this case, where a user didn't realize that a grid search would be occurring (and didn't want it to), the warnings will let them know. However, this would mean we would have to come up with a way to distinguish between when `grid_search=True` is explicitly specified vs. when it's not specified but the value is still `True` since that's the default. This is complicated to say the least and I am not sure that the warning is worth the added complexity. 

We will make sure to explicitly mention this change in the release notes and highlight it in the documentation as well.